R/kernel.r

```diff
@@ -101,11 +101,51 @@ handle_shell = function() {
         history(msg)
     } else if (msg$header$msg_type == "complete_request") {
         complete(msg)
+    } else if (msg$header$msg_type == "is_complete_request") {
+        is_complete(msg)
     } else {
         print(c("Got unhandled msg_type:", msg$header$msg_type))
     }
 },
 
+is_complete = function(request) {
+    code <- request$content$code
+    message <- tryCatch({
+        evaluate:::parse_all(code)
+        # the code compiles, so we are complete (either no code at all / only
+        # comments or syntactically correct code)
+        "complete" # no function, so no return!
+    }, warning = function(e){
+        return("warning") # IMO this can't happend...
```

Comment:
warnings aren't fatal, right? So if we don't know of a case where this can happen, maybe it's better to leave out the handler and let the warning print out.